planner/core/expression_rewriter.go

```diff
@@ -1292,3 +1296,86 @@ func (er *expressionRewriter) toColumn(v *ast.ColumnName) {
 	}
 	er.err = ErrUnknownColumn.GenWithStackByArgs(v.String(), clauseMsg[er.b.curClause])
 }
+
+func (er *expressionRewriter) evalDefaultExpr(v *ast.DefaultExpr) {
+	stkLen := len(er.ctxStack)
+	var colExpr *expression.Column
+	switch c := er.ctxStack[stkLen-1].(type) {
+	case *expression.Column:
+		colExpr = c
+	case *expression.CorrelatedColumn:
+		colExpr = &c.Column
+	default:
+		colExpr, er.err = er.schema.FindColumn(v.Name)
+		if er.err != nil {
+			er.err = errors.Trace(er.err)
+			return
+		}
+		if colExpr == nil {
+			er.err = ErrUnknownColumn.GenWithStackByArgs(v.Name.OrigColName(), "field_list")
+			return
+		}
+	}
+	dbName := colExpr.DBName
+	if dbName.O == "" {
+		// if database name is not specified, use current database name
+		dbName = model.NewCIStr(er.ctx.GetSessionVars().CurrentDB)
+	}
+	if colExpr.OrigTblName.O == "" {
```

Comment:
What about `select default(c) from (select a as c from t) as t0`? Has `c` been resolved as `t.a` already?

Reply:
There is a test for this case, please see the comment I left at the test code.